### PR TITLE
Create kadena app - base setup

### PR DESCRIPTION
**Pull request by jermaine150**
Create Kadena App - project generator with `@kadena/client` integration and contract types generation with `@kadena/pactjs-cli`.

Has support for NextJs now. VueJs and Angular templates will be added later. The Nextjs template project can be found in tools/create-kadena-app/templates/nextjs. The template projects aren't part of the rush monorepo nor the build.

For more context, (and also for reviewing purposes)  check out the rich-diff of the [README.md](https://github.com/kadena-community/kadena.js/pull/147/files?short_path=81fa9ba#diff-81fa9ba6d3768719f90f182ae379bbb7cbd7428be2dc74fe397b97f4704e7c3f)

## Check off the following:

- [X] I have reviewed my changes and run the appropriate tests.
- [X] I have have run `rush change` to add the appropriate change logs.
- [X] I have added/edited docs.
- [X] I have added tutorials.
- [X] I have double checked and DEFINITELY added docs.

### CODE REVIEW

File: packages/tools/create-kadena-app/src/generate-project/index.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+      if (NOTIMPLEMENTED_PROJECT_TEMPLATES.includes(template)) {
+        program.error(`Support for "${template}" is coming soon`);
+      }
+
+      if (!isValidProjectName(name)) {
+        program.error(`value "${name}" is an invalid project name`);
+      }
+
+      console.log('dirname', __dirname);
```
Comment (nillo) [inline packages/tools/create-kadena-app/src/generate-project/index.ts:175]: do we need this logs

File: packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import Head from 'next/head'
+import React, { useState } from 'react'
+import { SpinnerRoundFilled } from 'spinners-react'
+import { Pact, signWithChainweaver } from '@kadena/client'
+import styles from '../styles/Home.module.css'
```
Comment (nillo) [inline packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx:5]: Maybe home should be in lowercase (pref)

File: packages/tools/create-kadena-app/templates/nextjs/src/styles/Home.module.css [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  color: inherit;
+  text-decoration: none;
+  border: 1px solid #eaeaea;
+  border-radius: 10px;
+  transition: color 0.15s ease, border-color 0.15s ease;
+  width: 100%;
+  max-width: 1000px;
+}
+
+.cardtitle {
```
Comment (nillo) [inline packages/tools/create-kadena-app/templates/nextjs/src/styles/Home.module.css:36]: maybe use camelCasing (pref)

File: packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+const accountKey = (account: string): string => account.split(':')[1]
+
+const Home: React.FC = (): JSX.Element => {
+  const [account, setAccount] = useState<string>('')
+  const [messageToWrite, setMessageToWrite] = useState<string>('')
+  const [messageFromChain, setMessageFromChain] = useState<string>('')
+  const [writeInProgress, setWriteInProgress] = useState<boolean>(false)
+
+  const handleAccountInputChange = (event: any): void => {
```
Comment (nillo) [inline packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx:19]: `
const handleAccountInputChange = (event: React.ChangeEvent<HTMLInputElement>)
`

File: packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    setAccount(event.target.value)
+  }
+
+  const handleWriteMessageInputChange = (event: any): void => {
```
Comment (nillo) [inline packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx:23]: `const handleWriteMessageInputChange = (event: React.ChangeEvent<HTMLInputElement>)`

File: packages/tools/create-kadena-app/templates/nextjs/src/styles/Home.module.css [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  margin: 0.5rem 0;
+}
+
+.cardsection {
```
Comment (nillo) [inline packages/tools/create-kadena-app/templates/nextjs/src/styles/Home.module.css:40]: idem

File: packages/tools/create-kadena-app/templates/nextjs/src/styles/Home.module.css [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+}
+
+.cardsection input, .cardsection textarea {
```
Comment (nillo) [inline packages/tools/create-kadena-app/templates/nextjs/src/styles/Home.module.css:46]: idem

**Review by nillo — APPROVED** [review]
Some minor feedback, other than that lgtm

File: packages/tools/create-kadena-app/src/generate-project/index.ts (hunk 1 above)
Comment (jermaine150) [reply packages/tools/create-kadena-app/src/generate-project/index.ts]: Good catch

**Review by nillo — APPROVED** [review]

File: packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
+import Head from 'next/head'
+import React, { useState } from 'react'
+import { SpinnerRoundFilled } from 'spinners-react'
+import { Pact, signWithChainweaver } from '@kadena/client'
+import styles from '../styles/main.module.css'
+
+const NETWORK_ID = 'testnet04'
```
Comment (Randynamic) [inline packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx:7]: Can we make this build configurable
e.g. `pnpm run build --production` will change this to mainnet or at least look into a `kadena.config` file.

We can of course add this in a later feature.

**Review by Randynamic — APPROVED** [review]

File: packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx (hunk 8 above)
Comment (jermaine150) [reply packages/tools/create-kadena-app/templates/nextjs/src/pages/index.tsx:7]: Yes, the current contract is only deployed on testnet, but I can atleast make it configurable and move to config files. Will do that in the PR for the VueJS template I'm currently working on.

**Review by nillo — APPROVED** [review]

**Review by Randynamic — APPROVED** [review]

**Pull request merged**